### PR TITLE
executor: support upgrader

### PR DESCRIPTION
**Pull request by arlottang**
### What problem does this PR solve?

-  Refactoring the logic of Executor SyncStatefulSet, splitting it into smaller modules
-  Support dynamic binding and deletion of PVCs and PVs in the process of creating Executor StatefulSet
-  For Executor Upgrade, basic functions implemented only, subsequent to be optimized
- Support sync Executor status, but not volumes and members
-  Modify go sum

Issue link:

### What is changed and how it works?

- Deploy CR files with stateful tag to control PVC and PV
- Note: Basic functions implemented only, subsequent to be optimized

### Code changes

- [x] Has Go code change
- [ ] Has CI related scripts change

### Check List  

Tests 

- [ ] Unit test
- [ ] E2E test
- [ ] Stability test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Breaking backward compatibility
- [ ] Other side effects:

### CODE REVIEW

File: pkg/manager/member/executor_manager.go [hunk 1]
```diff
@@ -28,18 +29,22 @@ const (
 	executorPort = 10241
 	// tiflowExecutorDataVolumeMountPath is the mount path for tiflow-executor data volume
 	tiflowExecutorDataVolumeMountPath = "/etc/tiflow-executor"
+	// tiflowExecutorStorageVolumeMountPath is the mount path for tiflow-executor storage volume
+	tiflowExecutorStorageVolumeMountPath = "/tmp/tiflow-executor"
```
Comment (lichunzhu) [inline pkg/manager/member/executor_manager.go]: It's a bit strange to use `/tmp` directory as data dir here. Maybe `/var/lib` or something else?

File: pkg/manager/member/executor_manager.go [hunk 2]
```diff
@@ -312,23 +317,136 @@ func (m *executorMemberManager) getNewExecutorStatefulSetForTiflowCluster(ctx co
 		return nil, fmt.Errorf("config-map for tiflow-exeutor is not found, tifloeCluster %s/%s", tc.Namespace, tc.Name)
 	}
 
-	executorConfigMap := cfgMap.Name
+	// todo: Need to handle the secret if it is existed
```
Comment (lichunzhu) [inline pkg/manager/member/executor_manager.go]:
```suggestion
	// todo: Need to handle the secret if it is exists
```

**Comment by arlottang** [comment]
Wait for the next commit and then merge them together

File: pkg/manager/member/executor_manager.go [hunk 3]
```diff
@@ -445,56 +517,58 @@ func (m *executorMemberManager) getNewExecutorStatefulSetForTiflowCluster(ctx co
...
+	}
+
+	// get Annotation mount info, and add it
+	annoMount, _ := annotationsMountVolume()
+	volMounts = append(volMounts, annoMount)
+
+	// todo: Need to be modified soon
+	// handle pvc mount, and add it
+	pvcMount := corev1.VolumeMount{
+		Name:      DefaultStorageName,
```
Comment (lichunzhu) [inline pkg/manager/member/executor_manager.go:580]: Use another specilized name?

File: pkg/manager/member/executor_manager.go (hunk 1 above)
Comment (arlottang) [reply pkg/manager/member/executor_manager.go]: Fine, test only, waiting for a discussion to decide where to set.
To be modified on the next pr

File: pkg/manager/member/executor_manager.go (hunk 3 above)
Comment (arlottang) [reply pkg/manager/member/executor_manager.go:580]: To be modified on the next

File: pkg/manager/member/executor_upgrader.go [hunk 4]
```diff
@@ -1 +1,111 @@
...
+					ns, tcName, podName)
+			}
+			// todo: Need to be modified
+			if _, exist := tc.Status.Executor.Members[podName]; !exist {
+				return controller.RequeueErrorf("tiflowCluster: [%s/%s]'s upgrade tiflow-executor pod: [%s] is not exist",
+					ns, tcName, podName)
+			}
+			continue
+		}
+		mngerutils.SetUpgradePartition(newSts, i)
```
Comment (lichunzhu) [inline pkg/manager/member/executor_upgrader.go:100]: maybe leave a todo here.
Need to re-arrange this executor's tasks in the future.

**Review by lichunzhu — COMMENTED** [review]
Rest LGTM

File: pkg/manager/member/executor_upgrader.go (hunk 4 above)
Comment (arlottang) [reply pkg/manager/member/executor_upgrader.go:100]: ok

**Review by lichunzhu — APPROVED** [review]

**Pull request merged**